internal/common/client.go

```diff
@@ -14,6 +14,10 @@ type Client interface {
 	// StartOfTurnUpdate is where SOMASServer.updateIsland sends the game state over
 	// at start of turn. Do whatever you like here :).
 	StartOfTurnUpdate(gameState GameState)
+
+	// GameStateUpdate updates game state mid-turn.
+	GameStateUpdate(gameState GameState)
```

Comment:
Just wondering, why do we need different start of turn and mid-turn updates?

Reply:
Because we're dispatching actions and mutations for `gameState` immediately (e.g. IIGO deducts salary at start of IIGO from the common pool); therefore, the state is updated by IIGO mid-turn.

Reply:
We can technically get rid of `StartOfTurnUpdate`, but I feel a good distinction for clients to know when the turn is started is good, disasters also happen after all the orgs and end of turn actions so that is where they will know if a disaster happened.

Further, that's also where other clients can learn what other end of turn actions taken by other clients.

Reply:
Ah right, ok - thanks for explaining :)